app/src/main/java/com/communistutopia/spacetrader/model/Universe.kt

```diff
@@ -0,0 +1,78 @@
+package com.communistutopia.spacetrader.model
+
+import kotlin.random.Random
+import kotlin.math.pow
+import kotlin.math.sqrt
+
+/**
+ * Singleton object that handles all the logic for generating universe and other events
+ */
+object Universe {
+
+    // Constants
+    private val X_SIZE: Int = 150
+    private val Y_SIZE: Int = 100
+    private val NUM_SYSTEMS: Int = 10
+
+    private val solarSystems: ArrayList<SolarSystem> = ArrayList()
+    private val universe: HashMap<Pair<Int, Int>, String> // Initializing a 2D array is the most frustrating ordeal. So this is a hacky solution
+    private val random: Random
+
+
+    init {
+        universe = HashMap()
+        random = Random.Default
+    }
+
+    fun generateUniverse() {
```

Comment:
Also  please javadoc or kotlin doc, so we won't have to come back and do it, then it's also easier to see exactly what is happening in the method :)